src/backend/routes/user/edit.user.ts

```diff
@@ -0,0 +1,43 @@
+import { Request, Response } from 'express'
+import { StatusCodes } from 'http-status-codes'
+import { prisma } from '../../database'
+import { TRoute } from '../../routes/types'
+import { handleRequest } from '../../utils/request.utils'
+import { authorize } from '../../utils/middleware.utils'
+import {body} from 'express-validator'
+
+
+export default {
+    method: 'patch',
+    path: '/api/user/edit',
+    validators: [
+        authorize,
+        body().custom(body => {
+            const keys = Object.keys(body)
+            const validUpdates = [ 'userId','email', 'name', 'surname', 'address', 'birthDate', 'idCardNumber']
+            return keys.every(key => validUpdates.includes(key))
```

Comment:
would be great to validate also format of this data, like if the email is actually email, and brithdDate is a date etc, express-validator have such functions, you can do it in another file, to have helper calls that can be also used for validation in registration of users